tests/accumulate-headers-after-304.js

```diff
@@ -0,0 +1,212 @@
+        let updateField = new Field(updateHeaderName, 'x');
+        updateField.finalize();
+        const updateHeaderLength = updateField.raw().length;
+        
+        {
+            const prefixSize = this.maxPrefixSize() - updateHeaderLength;
+            let testCase = new HttpTestCase(`forward a cachable response with a prefix size less than the maximum allowed ${prefixSize}<${this.maxPrefixSize()}`);
+            testCase.client().request.for(resource);
+            if (Config.smp())
+                testCase.client().nextHopAddress = this._workerListeningAddresses[1];
+            testCase.server().serve(resource);
+            testCase.server().response.enforceMinimumPrefixSize(prefixSize);
+            testCase.server().response.header.add(hitCheck);
+            testCase.client().checks.add((client) => {
+                client.expectStatusCode(200);
+            });
+            await testCase.run();
+        }
+
+        {
+            let testCase = new HttpTestCase('respond with a 304 hit');
+            testCase.client().request.for(resource);
+            if (Config.smp())
+                testCase.client().nextHopAddress = this._workerListeningAddresses[1];
+            testCase.client().request.conditions({ ims: resource.notModifiedSince() });
+            testCase.client().checks.add((client) => {
+                client.expectStatusCode(304);
+            });
+            await testCase.run();
+        }
+
+        let updatingResponse = null;
+        {
+            let testCase = new HttpTestCase(`attempt to increase the cached prefix size up to the maximum allowed: ${this.maxPrefixSize()}`);
+
+            resource.modifyNow();
+            resource.expireAt(FuzzyTime.DistantFuture());
+
+            testCase.client().request.for(resource);
+            testCase.client().request.conditions({ ims: resource.modifiedSince() });
+            testCase.client().request.header.add("Cache-Control", "max-age=0");
+            if (Config.smp())
+                testCase.client().nextHopAddress = this._workerListeningAddresses[2];
+
+            testCase.server().response.header.add(updateField);
+            testCase.server().response.startLine.code(304);
+            testCase.server().serve(resource);
+            testCase.check(() => {
+                updatingResponse = testCase.server().transaction().response;
+                const receivedResponse = testCase.client().transaction().response;
+                assert(receivedResponse.startLine.codeInteger() === 200);
+                assert.equal(receivedResponse.header.value(hitCheck.name), hitCheck.value, "preserved originally cached header field");
+            });
+            await testCase.run();
+
+            // need some time to update the cached entry for disk-only SMP configurations
+            // without it the initially cached entry may be returned
+            // TODO: a better way to achieve this?
+            await Gadgets.SleepMs(1000);
```

Comment:
If waiting for the caching operations to end works, then please use that instead of sleeping:

```suggestion
            await this.dut.finishCaching();
```

If there are other, similar inter-case dependencies in this test, please wait to finishCaching() there as well.

Reply:
I am glad you found a decent workaround, allowing this PR to move forward! The `mgr:openfd_objects` problem became SQUID-892.